src/v/cluster/types.h

```diff
@@ -1003,6 +1003,11 @@ struct remote_topic_properties
     operator<<(std::ostream&, const remote_topic_properties&);
 };
 
+struct remote_partition_properties {
+    s3::bucket_name bucket;
```

Comment:
bucket here suggests that it is applicable in every case, but for recovery we don't use it which is a bit misleading. `read_replica_bucket`?